cranelift/codegen/src/isa/pulley_shared/inst/args.rs

```diff
@@ -122,6 +122,12 @@ newtype_of_reg!(XReg, WritableXReg, RegClass::Int);
 newtype_of_reg!(FReg, WritableFReg, RegClass::Float);
 newtype_of_reg!(VReg, WritableVReg, RegClass::Vector);
 
+impl XReg {
+    /// Index of the first "special" register, or the end of which registers
+    /// regalloc is allowed to use.
+    pub const SPECIAL_START: u8 = pulley_interpreter::regs::XReg::SPECIAL_START;
```

Comment:
Do we have constants for sp and fp? Can we define them like `SPECIAL_START + 0` and `SPECIAL_START + 1` to ensure this stays in sync?

Reply:
Does [this test](https://github.com/bytecodealliance/wasmtime/pull/9666/files#diff-577507af7de53d99c50170886b2c15e77f63375efaeb450058275f4eae3fc1d9R103) do what you're thinking?

Reply:
That tests the other direction: it asserts that every register from `SPECIAL_START` onward is special, but it doesn't test that every special register is beyond `SPECIAL_START`. I guess I'm suggesting a different way to define fp and sp, where we use the constant (plus a nonnegative offset) directly to have this by construction.

Reply:
Ah good point! The way these are defined in pulley is a bit special right now where each register is an `enum XReg { x0, ... sp, ... }` for example. That's uniform across x/f/v registers so I'm wary to change that much. That being said I'll enhance the test to assert that everything before `SPECIAL_START` is non-special.

Reply:
Looks good!